### PR TITLE
refactors encryption key management

### PR DESCRIPTION
**Pull request by xh3b4sd**
This is properly tested on ginger.

### CODE REVIEW

File: service/awsconfig/cluster_framework.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 	"github.com/giantswarm/operatorkit/client/k8scrdclient"
 	"github.com/giantswarm/operatorkit/framework"
 	"github.com/giantswarm/operatorkit/informer"
+	"github.com/giantswarm/randomkeys"
 	"github.com/giantswarm/randomkeytpr"
```
[the remark this answers is not in the document]
Comment (kopiczko) [reply service/awsconfig/cluster_framework.go]: Why? I don't think semantics changed. It's just the interface and k8s lib versions.
@jgsqware can you confirm?

[the remark this answers is not in the document]
Comment (xh3b4sd) [reply service/awsconfig/cluster_framework.go]: Ok cool. I got confused about the TPR in the project name. Checking the deps you are right. Should be good with updating vendor deps there. Cool.

File: service/awsconfig/v10/cloudconfig/cloud_config.go [hunk 2]
```diff
@@ -68,8 +63,9 @@ func New(config Config) (*CloudConfig, error) {
 	}
 
 	newCloudConfig := &CloudConfig{
-		// Dependencies.
-		logger:          config.Logger,
+		kmsClient: config.KMSClient,
```
[the remark this answers is not in the document]
Comment (rossf7) [reply service/awsconfig/v10/cloudconfig/cloud_config.go]: I think this is a big improvement. The KMS encryption is a required step for rendering cloudconfig on AWS. So the cloudconfig service should handle it.

File: service/awsconfig/v10/cloudconfig/cloud_config_test.go [hunk 3]
```diff
@@ -8,18 +8,19 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/aws/aws-sdk-go/service/kms"
 	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/certs/legacy"
 	"github.com/giantswarm/micrologger/microloggertest"
-	"github.com/giantswarm/randomkeytpr"
+	"github.com/giantswarm/randomkeys"
```
[the remark this answers is not in the document]
Comment (rossf7) [reply service/awsconfig/v10/cloudconfig/cloud_config_test.go]: Great that randomkeys is used now. The older resources will eventually get deleted keeping randomkeytpr till then is fine IMO.

File: service/awsconfig/v10/cloudconfig/cloud_config_test.go [hunk 4]
```diff
@@ -84,8 +85,8 @@ func Test_Service_CloudConfig_NewMasterTemplate(t *testing.T) {
 		})
 
 		t.Run("VerifyAPIServerEncryptionKey", func(t *testing.T) {
-			if !strings.Contains(decoded, "fekhfiwoiqhoifhwqefoiqwefoikqhwef") {
-				t.Fatalf("expected %#v got %#v", "cloud config to contain apiserver encryption key", "none")
+			if !strings.Contains(decoded, "H4sIAAAAAAAA/1SNMQ7CMAxF957CF+jQNSviCuwldYgVYTd2aBQh7o4CVREeLL33pf8T8eLgzF7bWkj4JBzoNswrXVCNhB1s06Bo8lCP5gaAEf6wC0OvWOxDq8pGC+oRzmj+6r/UL2GzH43A8x1dt9MhYW90EDDFQFUoR6EQa8YglGv/KceKYR+hBblQaQ6er3cAAAD//9QjGEbUAAAA") {
```
[the remark this answers is not in the document]
Comment (rossf7) [reply service/awsconfig/v10/cloudconfig/cloud_config_test.go]: Even harder to read 😄 but nice that the test confirms the value is encrypted.

File: service/awsconfig/v10/version_bundle.go [hunk 5]
```diff
@@ -19,6 +19,11 @@ func VersionBundle() versionbundle.Bundle {
 				Description: "Masked systemd-networkd-wait-online unit.",
 				Kind:        versionbundle.KindChanged,
 			},
+			{
+				Component:   "cloudconfig",
+				Description: "Improved encryption key injection.",
+				Kind:        versionbundle.KindChanged,
+			},
```
Comment (kopiczko) [inline service/awsconfig/v10/version_bundle.go]: I'd say this is a bugfix for aws-operator. Previous release was passing the key insecurely.

Comment (xh3b4sd) [reply service/awsconfig/v10/version_bundle.go]: Done.

**Comment by xh3b4sd** [comment]
Successfully tested via integration tests. At least through those who do not lack via the AWS performance.

**Comment by xh3b4sd** [comment]
Successfully tested on ginger.

**Pull request merged**